packages/datatrak-web/src/features/Survey/utils/usePrimaryEntityQuestionAutoFill.ts

```diff
@@ -4,26 +4,47 @@
  */
 import keyBy from 'lodash.keyby';
 import { useCurrentUserContext, useEntityAncestors } from '../../../api';
-import { useSurveyForm } from '../SurveyContext';
-import { getAllSurveyComponents, getPrimaryQuestionAncestorAnswers } from './utils';
-import { Entity } from '@tupaia/types';
+import { getParentQuestionId } from './utils';
+import { SurveyScreenComponent, Entity } from '../../../types';
+
+// Get the parent question ancestors recursively for the primary entity question
+export const getEntityQuestionAncestorAnswers = (
+  question: SurveyScreenComponent,
+  questionsById: Record<string, SurveyScreenComponent>,
+  ancestorsByType: Record<string, Entity>,
+): Record<string, string> => {
+  const answer = ancestorsByType[question?.config?.entity?.filter?.type?.[0] ?? ''];
+  if (!answer) return {};
+
+  const parentQuestionId = getParentQuestionId(question);
+  const parentQuestion = parentQuestionId ? questionsById[parentQuestionId] : null;
+
+  const record = { [question.id as string]: answer.id };
```

Comment:
Out of interest, would ts ever think the question id was any type other than string?

Reply:
Yes it comes in as Question['code'] which throws a warning